src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Hover/IHoverInfoService.cs

```diff
@@ -9,4 +11,6 @@ namespace Microsoft.AspNetCore.Razor.LanguageServer.Hover;
 internal interface IHoverInfoService
```

Comment:
This was a leaky abstraction before, but somehow feels worse now. What about renaming the methods to `GetRazorHoverInfoAsync` and `TranslateDelegatedHoverInfoAsync` to make things a little bit nicer.

Reply:
@davidwengier: Out of curiosity, is there a reason to include `Razor` in `GetRazorHoverInfoAsync`? I'm just trying to understand the naming conventions, because they seem to be applied somewhat differently throughout the code base. I was always assuming that the lack of a language name meant it was for Razor, but HTML or C# specific APIs would state the name.

Reply:
Very good question/point. Here is my thinking on the topic (which i'll preface by pre-disclaiming that naming is hard):

The `HoverEndpoint` (probably) used to be called `DefaultRazorHoverEndpoint` and in my opinion `Default` is redundant because there is only one, and `Razor` is redundant because it refers to Razor *files*, and we only have endpoints that deal with Razor files (this somewhat didn't used to be true, so I don't blame the original authors or anything 🙂).

In the method name I'm suggesting here `Razor` refers not to the file type, but the language. So the `GetHoverInfoAsync` method only returns information if the hover happens at a position in the file that is in a Razor language construct. It doesn't delegate to C# or Html, and I think calling that out in the method name is a useful thing to do.

Having said that, naming is hard and the fact that the word "Razor" could mean a file type or a language can definitely make it a confusing stance. I would have no push back if it was preferred to keep the method name as is, and use a XML comment to call out that there is delegation happening in the method.

Reply:
> In the method name I'm suggesting here Razor refers not to the file type, but the language. So the GetHoverInfoAsync method only returns information if the hover happens at a position in the file that is in a Razor language construct. It doesn't delegate to C# or Html, and I think calling that out in the method name is a useful thing to do.

The makes sense to me. I didn't realize this wasn't delegating.

Reply:
Renaming, and also changing to `IHoverService` instead of `IHoverInfoService`